internal/controller/pipelinerun_controller.go

```diff
@@ -41,15 +43,83 @@ var (
 
 // PipelineRunReconciler reconciles a PipelineRun object
 type PipelineRunReconciler struct {
-	Client client.Client
-	Scheme *runtime.Scheme
-	Config *config.ControllerConfig
+	Client     client.Client
+	Scheme     *runtime.Scheme
+	Config     *config.ControllerConfig
+	KiteClient *kite.Client
 }
 
 func (r *PipelineRunReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := ctrllog.FromContext(ctx).WithName("PipelineRunController")
+	ctx = ctrllog.IntoContext(ctx, log)
+
+	// Get the PipelineRun object
+	var pipelineRun tektonv1.PipelineRun
+	if err := r.Client.Get(ctx, req.NamespacedName, &pipelineRun); err != nil {
+		// PipelineRun was deleted, nothing to do
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// Only process completed PipelineRuns
+	if !pipelineRun.IsDone() {
+		return ctrl.Result{}, nil
+	}
+
+	// Check if the PipelineRun failed
+	condition := pipelineRun.Status.GetCondition(apis.ConditionSucceeded)
+	if condition != nil && !condition.IsTrue() {
```

Comment:
I'd split this in a way that if `condition` is nil, use early return and then do
```go
if condition.IsTrue() {}
else {}
```

to make it more readable.